src/panels/config/scene/ha-scene-editor.ts

```diff
@@ -0,0 +1,727 @@
+                `
+              : ""}
+          </div>
+        </div>
+        <ha-fab
+          slot="fab"
+          ?is-wide="${!this.narrow}"
+          ?dirty="${this._dirty}"
+          icon="hass:content-save"
+          .title="${this.hass.localize("ui.panel.config.scene.editor.save")}"
+          @click=${this._saveScene}
+          class="${classMap({
+            rtl: computeRTL(this.hass),
+          })}"
+        ></ha-fab>
+      </ha-app-layout>
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+
+    const oldscene = changedProps.get("scene") as SceneEntity;
+    if (
+      changedProps.has("scene") &&
+      this.scene &&
+      this.hass &&
+      // Only refresh config if we picked a new scene. If same ID, don't fetch it.
+      (!oldscene || oldscene.attributes.id !== this.scene.attributes.id)
+    ) {
+      this._loadConfig();
+    }
+
+    if (changedProps.has("creatingNew") && this.creatingNew && this.hass) {
+      this._dirty = false;
+      this._config = {
+        name: this.hass.localize("ui.panel.config.scene.editor.default_name"),
+        entities: [],
+      };
+    }
+
+    if (changedProps.has("_entityRegistryEntries")) {
+      for (const entity of this._entityRegistryEntries) {
+        if (
+          !entity.device_id ||
+          IGNORED_DOMAINS.includes(computeDomain(entity.entity_id))
+        ) {
+          continue;
+        }
+        if (!(entity.device_id in this._deviceEntityLookup)) {
+          this._deviceEntityLookup[entity.device_id] = [];
+        }
+        if (
+          !this._deviceEntityLookup[entity.device_id].includes(entity.entity_id)
+        ) {
+          this._deviceEntityLookup[entity.device_id].push(entity.entity_id);
+        }
+      }
+    }
+  }
+
+  private _showMoreInfo(ev: Event) {
+    const entityId = (ev.currentTarget as any).entity;
+    fireEvent(this, "hass-more-info", { entityId });
+  }
+
+  private async _loadConfig() {
+    try {
+      const config = await getSceneConfig(
+        this.hass,
+        this.scene!.attributes.id!
+      );
+
+      if (!config.entities) {
+        config.entities = {};
+      }
+
+      this._entities = Object.keys(config.entities);
+
+      this._entities.forEach((entity) => {
+        this._storeState(entity);
+      });
+
+      const filteredEntityReg = this._entityRegistryEntries.filter(
+        (entityReg) => this._entities.includes(entityReg.entity_id)
+      );
+
+      for (const entityReg of filteredEntityReg) {
+        if (!entityReg.device_id) {
+          continue;
+        }
+        if (!this._devices.includes(entityReg.device_id)) {
+          this._devices = [...this._devices, entityReg.device_id];
+        }
+      }
+
+      activateScene(this.hass, this.scene!.entity_id);
+
+      // we want to check if states changed after the initial scene was activated, when is that done?
+      setTimeout(async () => {
```

Comment:
Why only start listening to this after 500 ms ? 

The best way to do this is to use the context ID. All states that are impacted and change will have the same ID as the service call. We don't currently return any result when we call a service, but that is something that we could add.